ui: add all jobs to jobs page filter

otan opened the pull request:
<img width="604" alt="image" src="https://user-images.githubusercontent.com/3646147/163288704-498f0de7-699a-48db-8667-9c7d63b0f8ef.png">

Resolves #79915

Release note (ui change): More jobs now shows up as a types filter
on the jobs page.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/79936)
<!-- Reviewable:end -->

otan commented:
hmm, actually with this added, the new jobs types don't seem to appear

<img width="943" alt="image" src="https://user-images.githubusercontent.com/3646147/163290261-f905bb84-29b1-4c1d-846d-2b05a61914cd.png">

maryliag reviewed (commented):
if you check directly on `crdb_internal.jobs` do you see the jobs you expected to see? That is the table used to display this data

Also, are the new types consider automatic jobs? If so we, we hide them when you have "all" selected on the filter, but they show up if you select a specific value on the filter https://github.com/cockroachdb/cockroach/pull/77331

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/79936#-:-N-c_-B-DfFcRioreEZV:b-491a4v)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
> if you check directly on crdb_internal.jobs do you see the jobs you expected to see? That is the table used to display this data

yes

> Also, are the new types consider automatic jobs? If so we, we hide them when you have "all" selected on the filter, but they show up if you select a specific value on the filter https://github.com/cockroachdb/cockroach/pull/77331

they don't show up with that filter :(

otan commented:
ok nvm this works with `start-single-node` but not `demo`. seems like an existing problem tho.

maryliag reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/79936)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @maryliag and @otan)

___
*[pkg/ui/workspaces/db-console/src/views/jobs/index.tsx, line 57 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/79936#-N-e0Gg15XE-2PmeBPwT:-N-e0Gg15XE-2PmeBPwU:bnywdwu) ([raw file](https://github.com/cockroachdb/cockroach/blob/b0569cc41f19ef5fd7167b488e52307892c1dbb7/pkg/ui/workspaces/db-console/src/views/jobs/index.tsx#L57)):*
> ```XML
>   {
>     value: JobType.TYPEDESC_SCHEMA_CHANGE.toString(),
>     label: "Type Descriptor Schema Change",
> ```

nit: some labels you're using singular and others plural. Seems like the majority of existing ones we use plural, so you can align these new labels to also use plural
___
*[pkg/ui/workspaces/db-console/src/views/jobs/index.tsx, line 73 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/79936#-N-e-dXn8Y4AMVFwP9wY:-N-e-dXoBPe-mLd5Y3D3:b-xpdudy) ([raw file](https://github.com/cockroachdb/cockroach/blob/b0569cc41f19ef5fd7167b488e52307892c1dbb7/pkg/ui/workspaces/db-console/src/views/jobs/index.tsx#L73)):*
> ```XML
>   {
>     value: JobType.ROW_LEVEL_TTL.toString(),
>     label: "Row-level Time-to-live Deletions",
> ```

nit: according to Vy's comment on the issue, remove the "row-level"


<!-- Sent from Reviewable.io -->

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/79936)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @maryliag)

___
*[pkg/ui/workspaces/db-console/src/views/jobs/index.tsx, line 57 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/79936#-N-e0Gg15XE-2PmeBPwT:-N-vPs__BrS2OeemUmah:bstvmez) ([raw file](https://github.com/cockroachdb/cockroach/blob/b0569cc41f19ef5fd7167b488e52307892c1dbb7/pkg/ui/workspaces/db-console/src/views/jobs/index.tsx#L57)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

nit: some labels you're using singular and others plural. Seems like the majority of existing ones we use plural, so you can align these new labels to also use plural
</blockquote></details>

hmm, some yes some no. "Changefeed" is singular, "Span Config Reconciliations" and "Auto-Statistics Creations" sound a bit awkward compared to the singular. Can switch if you think it's better.


<!-- Sent from Reviewable.io -->

otan committed:
ui: add all jobs to jobs page filter

Release note (ui change): More jobs now shows up as a types filter
on the jobs page.

maryliag reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/79936#-:-N-xDkBg1jLshv4uaFFL:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @maryliag and @otan)

___
*[pkg/ui/workspaces/db-console/src/views/jobs/index.tsx, line 57 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/79936#-N-e0Gg15XE-2PmeBPwT:-N-xDiNSB1f-mODuER1h:brxp9sd) ([raw file](https://github.com/cockroachdb/cockroach/blob/b0569cc41f19ef5fd7167b488e52307892c1dbb7/pkg/ui/workspaces/db-console/src/views/jobs/index.tsx#L57)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

hmm, some yes some no. "Changefeed" is singular, "Span Config Reconciliations" and "Auto-Statistics Creations" sound a bit awkward compared to the singular. Can switch if you think it's better.
</blockquote></details>

Sounds good!


<!-- Sent from Reviewable.io -->

otan commented:
tftr!

bors r=maryliag
